types/d3-color/index.d.ts

```diff
@@ -27,6 +28,11 @@ export interface ColorCommonInstance {
     brighter(k?: number): this;
     darker(k?: number): this;
     rgb(): RGBColor;
+    /**
+     * Returns a hexadecimal string representing this color.
+     * If this color is not displayable, a suitable displayable color is returned instead. For example, RGB channel values greater than 255 are clamped to 255.
+     */
+   hex(): string;
 }
 
 export interface Color {
```

Comment:
This is the `Color` interface you should add `hex()`.

Reply:
Shouldn't  `brighter`, `darker`, and `rgb` be moved as well?

Reply:
Only `hex()` would be required to me moved to the abstract base class `Color`. See [Color define statement in source](https://github.com/d3/d3-color/blob/d5258a56a0699ddffd751ff79206cc1399ee7f89/src/color.js#L171). `brighter` and `darker` are actually implemented on the different color space classes.